src/gatt-database.c

```diff
@@ -2711,10 +2798,32 @@ static struct pending_op *acquire_write(struct external_chrc *chrc,
 	return NULL;
 }
 
+static struct client_io *
+client_notify_io_get(struct external_chrc *chrc, int fd, struct bt_att *att)
```

Comment:
This is duplicating the code of `client_write_io_get`, they could be unified if you pass the additional `struct queue **ios` parameter.